src/app.rs

```diff
@@ -569,6 +570,38 @@ impl<'a, 'b, T: State + 'a> ApplicationBuilder<'a, 'b, T> {
         self
     }
 
+    /// Add a given ECS bundle to the game loop.
+    ///
+    /// A bundle is a container for registering a bunch of ECS systems and their dependent
+    /// resources and components.
+    ///
+    /// # Parameters
+    ///
+    /// - `bundle`: The bundle to add
+    ///
+    /// # Returns
+    ///
+    /// This function returns ApplicationBuilder after it has modified it, this is
+    /// wrapped in a `Result`.
+    ///
+    /// # Errors
+    ///
+    /// This function creates systems and resources, which use any number of dependent
+    /// crates or APIs, which could result in any number of errors.
+    /// See each individual bundle for a description of the errors it could produce.
+    ///
+    pub fn with_bundle<B>(mut self, bundle: B) -> Result<Self>
+    where
+        B: for<'c> ECSBundle<'a, 'b, (&'c EventsLoop)>,
```

Comment:
RenderSystem is special as it the only one that need access to `EventsLoop` during building.
What should we do if there will be new bundle that will require some other data from `Application` to build?

Reply:
Yeah, it's not perfect, but I haven't figured out a good solution to that yet.

Reply:
Updated, the bundle trait now works on the ApplicationBuilder directly.